hr_timesheet_employee_allowed/models/account_analytic_line.py

```diff
@@ -0,0 +1,40 @@
+# Copyright (C) 2020 Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class AccountAnalyticLine(models.Model):
+    _inherit = 'account.analytic.line'
+
+    @api.model
+    def default_get(self, field_list):
+        res = super(AccountAnalyticLine, self).default_get(field_list)
+        if res.get('employee_id'):
+            res.pop('employee_id', None)
+        return res
+
+    timesheet_line_user_id = fields.Many2one(
+        'res.users', "Current timesheet user",
+        default=lambda self: self.env.user)
+
+    @api.onchange('timesheet_line_user_id')
+    def onchange_employee_id_line(self):
```

Comment:
We could have this as a static domain attribute in the user field.
It does mean a long expression with several "OR", one for each case.